### ISSUE TITLE
WGS Workflow not parsing in Pipeline Builder

### ISSUE THREAD

**Issue by jsotobroad** [epam/pipeline-builder#15]
There is a wdl we regularly run but it errors out with this message when trying to `Build`.
`Context initialization failed due to Error: In scatter body keys [workflowoutputs, meta, parametermeta, declaration] are not allowed`.

The wdl can be found here - https://github.com/broadinstitute/wdl/blob/develop/scripts/broad_pipelines/PublicPairedSingleSampleWf_170412.wdl

There are two scatters in that wdl
```
scatter (unmapped_bam in flowcell_unmapped_bams) {
  
    # Because of a wdl/cromwell bug this is not currently valid so we have to sub(sub()) in each task
    # String base_name = sub(sub(unmapped_bam, "gs://.*/", ""), unmapped_bam_suffix + "$", "")

    String sub_strip_path = "gs://.*/"
    String sub_strip_unmapped = unmapped_bam_suffix + "$"

    # QC the unmapped BAM 
    call CollectQualityYieldMetrics {
      input:
        input_bam = unmapped_bam,
        metrics_filename = sub(sub(unmapped_bam, sub_strip_path, ""), sub_strip_unmapped, "") + ".unmapped.quality_yield_metrics",
        disk_size = flowcell_small_disk,
        preemptible_tries = preemptible_tries
    }
...
```
and
```
scatter (subgroup in CreateSequenceGroupingTSV.sequence_grouping) {
    # Generate the recalibration model by interval
    call BaseRecalibrator {
      input:
        input_bam = SortAndFixSampleBam.output_bam,
        input_bam_index = SortAndFixSampleBam.output_bam_index,
        recalibration_report_filename = base_file_name + ".recal_data.csv",
        sequence_group_interval = subgroup,
        dbSNP_vcf = dbSNP_vcf,
        dbSNP_vcf_index = dbSNP_vcf_index,
        known_indels_sites_VCFs = known_indels_sites_VCFs,
        known_indels_sites_indices = known_indels_sites_indices,
        ref_dict = ref_dict,
        ref_fasta = ref_fasta,
        ref_fasta_index = ref_fasta_index,
        disk_size = agg_small_disk,
        preemptible_tries = agg_preemptible_tries
    }  
  }  
  
  # Merge the recalibration reports resulting from by-interval recalibration
  call GatherBqsrReports {
    input:
      input_bqsr_reports = BaseRecalibrator.recalibration_report,
      output_report_filename = base_file_name + ".recal_data.csv",
      disk_size = flowcell_small_disk,
      preemptible_tries = preemptible_tries
  }
...
```

**Comment by jsotobroad** [comment]
It seems like these lines in the scatter block might be causing the error
```
    String sub_strip_path = "gs://.*/"
    String sub_strip_unmapped = unmapped_bam_suffix + "$"
```

**Comment by sidoruka** [comment]
@jsotobroad , this is fixed in PR#25. Provided script now builds correctly, could you please recheck using: http://pb.opensource.epam.com:10000/ ?

Here is a schematic of an output generated from a WDL workflow, you've provided:
![15_full_workflow](https://cloud.githubusercontent.com/assets/20998291/26250318/5495b826-3cb2-11e7-8e1d-36a92db9a201.png)

**Comment by jsotobroad** [comment]
@sidoruka it looks like `ver 0.3.9` on http://pb.opensource.epam.com:10000/ looks like its defaulting me to trying to make a tool when I first load the page?  I can't actually write in my own wdl and build it because I cant rid of this tool builder overlay.

![screen shot 2017-05-22 at 8 32 05 am](https://cloud.githubusercontent.com/assets/13023616/26308923/2b2fe72c-3ec9-11e7-9f00-b0c175495851.png)

**Comment by sidoruka** [comment]
@jsotobroad , seems that a redeploy of an application was in progress when you were testing. Should work fine now.

**Comment by jsotobroad** [comment]
I have the best timing >< .  Thanks that seemed to be it, can do things normally now!

**Comment by jsotobroad** [comment]
Yup that looks about right, just a very hefty workflow.  It does take a little bit of time to build (~5 seconds) but that might just be expected with the size of this workflow.  I honestly forgot how big it was until I saw this visualization

@vdauwera check it out!

**Comment by vdauwera** [comment]
Oh sweet! Yeah we have some massive workflows. And wait 'til we start using imports to combine this with others into mega-workflows...

I could't zoom in enough to see the details so I can't comment on exactness, but overall it looks very reasonable.

**Comment by jsotobroad** [comment]
Yea I just went to the dev url and tried it myself to look at each task/scatter

**Comment by vdauwera** [comment]
Oh didn't realize I could go to the url (doh). Looks awesome! 

Maybe there could be a modal to display a "building..." message, so that for big workflows like this people don't wonder if the program is working?

**Comment by sidoruka** [comment]
@vdauwera , thanks for the suggestion. We've added "Building..." indicator. Now it seems to be more user-friendly for large workflows.

![building](https://cloud.githubusercontent.com/assets/20998291/26646367/fc8e9e90-4643-11e7-997a-aab2e92b80a9.png)

**Comment by jsotobroad** [comment]
That's awesome

On Wed, May 31, 2017 at 1:59 PM, Aleksandr Sidoruk <notifications@github.com
